allolib-wasm/include/al_WebGPURenderShader.hpp

```
@@ -0,0 +1,523 @@
+    /**
+     * Create a render pipeline with the standard 48-byte InterleavedVertex layout
+     * and depth testing enabled (Depth32Float, LessEqual).
+     */
+    void createWithVertexLayout(GraphicsBackend& backend, const char* wgslSource, int uniformSize) {
+        createInternal(backend, wgslSource, uniformSize, true);
+    }
+
+    /**
+     * Create a GPU buffer and upload vertex data.
+     */
+    WGPUBuffer uploadMesh(const void* interleavedData, size_t dataSize) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)